citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java

```diff
@@ -107,27 +169,118 @@ public IAuthorityEntry find(String id) {
         });
         return entries;
     }
-    
+
+    @Override
+    public Set<IAuthorityEntry> findByNameInDataset(String firstName, String lastName, String citationGroupId,
+            List<String> uris, int page, int pageSize) throws GroupDoesNotExistException {
+        Optional<CitationGroup> group = groupRepository.findById(new Long(citationGroupId));
+        if (!group.isPresent()) {
+            throw new GroupDoesNotExistException("Group with id " + citationGroupId + " does not exist.");
+        }
+        Pageable paging = PageRequest.of(page, pageSize);
+        List<Person> persons = personAuthorityRepository.findPersonsByCitationGroupAndNameLikeAndUriNotIn(
+                (ICitationGroup) group.get(), firstName, lastName, uris, paging);
+        Set<IAuthorityEntry> entries = new HashSet<>();
+        persons.forEach(p -> {
+            Optional<AuthorityEntry> optional = entryRepository.findById(p.getLocalAuthorityId());
+            if (optional.isPresent()) {
+                entries.add(optional.get());
+            }
+        });
+        return entries;
+    }
+
+    @Override
+    public Set<IAuthorityEntry> findByNameInDataset(String firstName, String lastName, String citationGroupId, int page,
+            int pageSize) throws GroupDoesNotExistException {
+        Optional<CitationGroup> group = groupRepository.findById(new Long(citationGroupId));
+        if (!group.isPresent()) {
+            throw new GroupDoesNotExistException("Group with id " + citationGroupId + " does not exist.");
+        }
+        Pageable paging = PageRequest.of(page, pageSize);
+        List<Person> persons = personAuthorityRepository
+                .findPersonsByCitationGroupAndNameLike((ICitationGroup) group.get(), firstName, lastName, paging);
+        Set<IAuthorityEntry> entries = new HashSet<>();
+        persons.forEach(p -> {
+            Optional<AuthorityEntry> optional = entryRepository.findById(p.getLocalAuthorityId());
+            if (optional.isPresent()) {
+                entries.add(optional.get());
+            }
+        });
+        return entries;
```

Comment:
this looks like a lot of duplicate code (basically just one line is different?) you can probably just make that one method with an if statement that decides which repo method should be called.